src/WidgetFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Widget;
+
+use Psr\Container\ContainerInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\Factory\Factory;
+use Yiisoft\Factory\FactoryInterface;
+
+final class WidgetFactory extends Factory
+{
+    private static ?FactoryInterface $factory = null;
```

Comment:
```suggestion
    private static ?FactoryInterface $factory;
```

Reply:
Nope. We have to do it. It is a typed property. If we won't initialize it, it will be in uninitialized state.